storagepool/storagepool.go

```diff
@@ -0,0 +1,51 @@
+package storagepool
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+	"github.com/rancher/convoy-agent/cattle"
+)
+
+var Commands = []cli.Command{
+	{
+		Name:  "storagepool",
+		Usage: "Start convoy-agent as a storagepool agent",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "sp-uuid",
+				Usage: "set the storage pool uuid",
+			},
+			cli.StringFlag{
+				Name:  "sp-rootdir",
+				Usage: "set the storage pool rootdir",
+				Value: ".root",
+			},
+		},
+		Action:    storagepoolAgent,
+		ShortName: "sp",
+	},
+}
+
+func storagepoolAgent(c *cli.Context) {
+	healthCheckInterval := c.GlobalInt("healthcheck-interval")
+	healthCheckBaseDir := c.GlobalString("healthcheck-basedir")
+	storagepoolUUID := c.String("sp-uuid")
```

Comment:
sp-uuid should be generated by this code and stored.  then read and used later if its a restart.  Rather not do this logic in the wrapper scripts but in go itself.